kernel/kernel-api/src/main/java/io/delta/kernel/Scan.java

```diff
@@ -173,6 +173,16 @@ public FilteredColumnarBatch next() {
         initIfRequired();
         ColumnarBatch nextDataBatch = physicalDataIter.next();
 
+        // Add partition columns
+        // NOTE: Adding partition columns must be the first thing we do with the data batch because
+        // it is the last thing we remove from the physicalReadSchema when creating a ScanStateRow.
```

Comment:
Sorry I don't exactly follow this? Why is this / what necessitated this change? Was it wrong before?

Reply:
I see; I think maybe we could make this comment a little more clear? maybe add that adding this back converts our batch to have schema=physicalSchema which may or may not contain metadata columns?